[docs] Add getting started page with project templates

planger opened the pull request:
Also restructures documentation in general to allow adopters proceeding more quickly to the getting started page.
Adds a place holder for model operations, which we will add soon.

Contributed on behalf of STMicroelectronics.

tortmayr commented on content/documentation/actionhandler/content.md:
```diff
@@ -8,15 +8,28 @@ title = "Actions & Action Handler"
+Besides, adopters can replace and extend existing, or add additional action handlers for standard or custom action types.
 
 To do that the following steps have to be performed
 
```
Intended for line 81:
> Each action specification has a unique “kind” and can optionally declare additional data properties We recommend

Missing dot:  ... declare additional data properties. We recommend ...

tortmayr replied:
This still seems to be unresolved, unless I missed it.

tortmayr reviewed (changes requested):
Thanks you Philip for this huge documentation improvement! I really like the 
restructuring and the new getting started section. This is a huge step forward in lowering the entry  barrier for people interested in GLSP .

I did an extensive review of all you changes and also quickly parsed the unchanged sections to see if we missed something there. In addition, I have double checked wether all links work as expected.

You can find my suggestions and comments in-line. 

In addition, to this in-line suggestions I noticed the following three issues:

### The `one-sentence-per-line` pattern for markdown files is not used consistently. 
While are following the pattern in most sections there are some cases where multiple sentences are part of one line. 

I think we should double check this and ensure that we use one line per sentence. This procuded much nicer git diffs. 

If have added comments regarding this to  the root index [content.md](https://github.com/eclipsesource/graphical-lsp-website/pull/53/files#diff-4c8a825efe112f54d69c1c18270485e34098649f15c66ce808406523cfc88a05) file but did not do it for other files to avoid `inline-comment pollution` :wink: 

### The node snippets regarding di configuration are outdated.
With https://github.com/eclipse-glsp/glsp-server-node/pull/15 the di configuration approach got changed. As a consequence many node snippets are now outdated and need to be updated

### In the `Actions & Action handler` overview the node snippets are not collapsed by default

Overall this is an amazing change! Great work!

planger commented on content/documentation/integrations/content.md:
```diff
@@ -0,0 +1,161 @@
+* If you aim at providing a VS Code extension or Theia plugin, shipping a node-based GLSP server is easier, because you don't need to worry whether your users have a JVM installed. They'll have a node runtime already, otherwise they couldn't run VS Code or Theia. If you have more control over your users runtime, e.g. because the editor will run in a container in the cloud anyway, this is less of  ...
+* Using one language (Typescript) for your GLSP server and client leads to a more homogeneous developer experience than having to work with two languages (Java and Typescript).
+* If you already have an editor implementation in Java, e.g. based on GMF, etc., or an EMF metamodel, building a Java-based server opens lots of opportunities for reusing your existing business logic in the GLSP server.
+* If you want to work with EMF source models directly from your GLSP server, e.g. because you want to benefit from its command stack implementation, change notification support, etc. and don't want or need a separate emf.cloud model server for model management, using Java on the GLSP server is clearly better.
```
I removed this part of the sentence.

planger commented on content/documentation/overview/content.md:
```diff
@@ -0,0 +1,52 @@
+
+A *GLSP Server* is a separate backend process and provides the language-specific smarts and editing capabilities.
+It communicates with a *GLSP Client* over JSON-RPC with an [extensible action protocol](https://github.com/eclipse-glsp/glsp/blob/master/PROTOCOL.md).
+Each diagram editor is considered to be a single *GLSP client*.
```
Thanks, I tried to clarify.

tortmayr commented:
I guess the issue regarding the `one-sentence-per-line` convention is also unresolved, but I'm also fine with not doing that now

planger commented:
> Thank you Philip. The changes look good to me. There is one comment that still seems to be unresolved: [#53 (comment)](https://github.com/eclipsesource/graphical-lsp-website/pull/53#discussion_r861372499).

Right, fixed.

> I guess the issue regarding the one-sentence-per-line convention is also unresolved, but I'm also fine with not doing that now

Yeah, I omitted that for now in the existing text. The new text that I added should largely respect that it.
